types/react-native/globals.d.ts

```diff
@@ -272,3 +272,29 @@ declare class URLSearchParams {
     keys(): IterableIterator<string>;
     values(): IterableIterator<string>;
 }
+
+interface WebSocketMessageEvent extends Event {
+    data?: any;
+}
+interface WebSocketErrorEvent extends Event {
+    message: string;
+}
+interface WebSocketCloseEvent extends Event {
+    code?: number;
+    reason?: string;
+}
+
+interface WebSocket extends EventTarget {
+    readonly readyState: number;
+    send(data: string | ArrayBuffer | ArrayBufferView | Blob): void;
+    close(code?: number, reason?: string): void;
+    onopen: (() => void) | null;
+    onmessage: ((event: WebSocketMessageEvent) => void) | null;
+    onerror: ((event: WebSocketErrorEvent) => void) | null;
+    onclose: ((event: WebSocketCloseEvent) => void) | null;
+}
+
+declare var WebSocket: {
+    prototype: WebSocket;
+    new (uri: string, protocols?: string | string[], options?: {headers: {[headerName: string]: string}}): WebSocket;
```

Comment:
In Flow, a question mark to the left of a type identifier means it can be `undefined` (optional) or `null`. The `options` object has optional keys and should allow for more keys than the few it hints at.

```suggestion
    new (uri: string, protocols?: string | string[] | null, options?: { headers?: { options?: string, [key: string]: any }, [key: string]: any } | null): WebSocket;
```

Reply:
Declaring WebSocket as a class works too.
I used type merging (interface + var) instead of class because this pattern is used all over this type declaration file. For example XMLHttpRequest and Headers are defined in this way.

Added optional keys to options. Allowed protocols and options to be null.

Reply:
Fair enough; still, thanks for updating!